packages/dataviews/src/dataviews-layouts/grid/preview-size-picker.tsx

```diff
@@ -3,108 +3,78 @@
  */
 import { RangeControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useMemo, useContext } from '@wordpress/element';
+import { useContext } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import DataViewsContext from '../../components/dataviews-context';
 import type { ViewGrid } from '../../types';
 
-const viewportBreaks: {
-	[ key: string ]: { min: number; max: number; default: number };
-} = {
-	xhuge: { min: 3, max: 6, default: 5 },
-	huge: { min: 2, max: 4, default: 4 },
-	xlarge: { min: 2, max: 3, default: 3 },
-	large: { min: 1, max: 2, default: 2 },
-	mobile: { min: 1, max: 2, default: 2 },
-};
-
-/**
- * Breakpoints were adjusted from media queries breakpoints to account for
- * the sidebar width. This was done to match the existing styles we had.
- */
-const BREAKPOINTS = {
-	xhuge: 1520,
-	huge: 1140,
-	xlarge: 780,
-	large: 480,
-	mobile: 0,
-};
-
-function useViewPortBreakpoint() {
-	const containerWidth = useContext( DataViewsContext ).containerWidth;
-	for ( const [ key, value ] of Object.entries( BREAKPOINTS ) ) {
-		if ( containerWidth >= value ) {
-			return key;
-		}
-	}
-	return 'mobile';
-}
-
-export function useUpdatedPreviewSizeOnViewportChange() {
-	const view = useContext( DataViewsContext ).view as ViewGrid;
-	const viewport = useViewPortBreakpoint();
-	return useMemo( () => {
-		const previewSize = view.layout?.previewSize;
-		let newPreviewSize;
-		if ( ! previewSize ) {
-			return;
-		}
-		const breakValues = viewportBreaks[ viewport ];
-		if ( previewSize < breakValues.min ) {
-			newPreviewSize = breakValues.min;
-		}
-		if ( previewSize > breakValues.max ) {
-			newPreviewSize = breakValues.max;
-		}
-		return newPreviewSize;
-	}, [ viewport, view ] );
-}
+const imageSizes = [
+	{
+		value: 230,
+		breakpoint: 1,
+	},
+	{
+		value: 290,
+		breakpoint: 1112, // at minimum image width, 4 images display at this container size
+	},
+	{
+		value: 350,
+		breakpoint: 1636, // at minimum image width, 6 images display at this container size
+	},
+	{
+		value: 430,
+		breakpoint: 588, // at minimum image width, 2 images display at this container size
```

Comment:
This tries to emulate the current behaviour, where setting the grid at its biggest size on a small container and resizing the window preserves the biggest size setting, even when more size options appear. This is probably not too important a point in user-facing terms though, because people don't go around resizing windows that much 😄